src/Microsoft.Health.TaskManagement/JobHosting.cs

```diff
@@ -204,6 +206,7 @@ await using (new Timer(async _ => await PutJobHeartbeatAsync(queueClient, jobInf
             }
         }
 
+        [Obsolete("Heartbeats should only update timestamp, results should only be written when job reaches terminal state.")]
```

Comment:
Still not sure on this. We update 1 field instead of 2. Still curious if this is more a design motive vs perf motive

Reply:
It is not about how many fields are updated, but about how many times entire job record in the Result column is update and with what data. If thus record contains all intermediate states N and each state is saved as soon as it arrived, it leads to N*N problem (that is how much data is saved overall). This should be discouraged, hence obsolete tag.

Reply:
> about how many times entire job record in the Result column is updated and with what data.

I think the preference here is that Result should not contain the definition, it should only be data or state created or obtained by the action of the job.

> that is how much data is saved overall, actually square(N)/2 to be exact). This should be discouraged, hence obsolete tag.

If Result was updated new distinct values at each logical job checkpoint -- so assuming result is only updated when you would potentially insert a new record. Why is this worse than inserting an entirely new record that also duplicates most of the fields of the previous record?

Reply:
The goal is never save distinct list of any values to a job, but rather each of those values separately. If we do that amount of data saved is proportional to N (not square(N)).